src/bin/stratisd.rs

```diff
@@ -78,6 +86,33 @@ fn handle_udev_add(event: &libudev::Event) -> Option<(PathBuf, Device)> {
     None
 }
 
+/// To ensure only one instance of stratisd runs at a time, acquire an
+/// exclusive lock. Exit if lock attempt fails.
+fn create_pid_file() -> StratisResult<File> {
+    let mut f = OpenOptions::new()
+        .read(true)
+        .write(true)
+        .create(true)
+        .open(STRATISD_PID_PATH)?;
+    match flock(f.as_raw_fd(), FlockArg::LockExclusiveNonblock) {
+        Ok(()) => {f.write(format!("{}\n", getpid()).as_bytes())?;
```

Comment:
Why the "\n"? Is that a rule for these files?

Reply:
seems to be what other .pid files do, from a quick survey on my local machine, yeah.